src/Workspaces/Core/Portable/Workspace/Solution/Project.cs

```diff
@@ -254,6 +256,19 @@ public bool ContainsAnalyzerConfigDocument(DocumentId documentId)
             return ImmutableHashMapExtensions.GetOrAdd(ref _idToAnalyzerConfigDocumentMap, documentId, s_createAnalyzerConfigDocumentFunction, this);
         }
 
+        public async ValueTask<IEnumerable<SourceGeneratedDocument>> GetSourceGeneratedDocumentsAsync(CancellationToken cancellationToken = default)
```

Comment:
`ValueTask<IEnumerabe>`?  WHy not `ValueTask<ImmutableArray>`?

Reply:
I was keeping this consistent with the other Documents/AdditionalDocuments/AnalyzerConfigDocuments, in that those always return IEnumerable.